backend/model/model.go

```diff
@@ -2109,6 +2110,14 @@ type LogAlertEvent struct {
 	SentAt     time.Time
 }
 
+type SavedSegment struct {
+	Model
+	Name       string
+	EntityType modelInputs.SavedSegmentEntityType `gorm:"index:idx_saved_segment"`
```

Comment:
Add an index here since most queries will be by project_id and entity_type (e.g. "Log saved segments for project 1", "Trace saved segments for Project 23")

Reply:
nit: can use the `priority` field in gorm so that project id is ordered before entity type - assuming sometime in the future we might find it useful to query all segments for all entity types in a project, but would never want to query segments for a single entity type across all projects https://gorm.io/docs/indexes.html#:~:text=You%20can%20specify%20the%20order%20with%20the%20priority%20option%2C%20the%20default%20priority%20value%20is%2010%2C%20if%20priority%20value%20is%20the%20same%2C%20the%20order%20will%20be%20based%20on%20model%20struct%E2%80%99s%20field%20index